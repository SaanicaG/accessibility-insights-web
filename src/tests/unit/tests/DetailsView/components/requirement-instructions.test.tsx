// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import {
    RequirementInstructions,
    RequirementInstructionsProps,
} from 'DetailsView/components/requirement-instructions';
import * as React from 'react';

describe('RequirementInstructions', () => {
    it('renders with content from props', () => {
        const props: RequirementInstructionsProps = {
            howToTest: <p>how to test stub</p>,
        };

        const renderResult = render(<RequirementInstructions {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
