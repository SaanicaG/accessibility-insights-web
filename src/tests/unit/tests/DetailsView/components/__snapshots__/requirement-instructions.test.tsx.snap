// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequirementInstructions renders with content from props 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <button
      aria-expanded="true"
      class="ms-Button ms-Button--action ms-Button--command collapsible root-109"
      data-is-focusable="true"
      type="button"
    >
      <span
        class="ms-Button-flexContainer flexContainer-110"
        data-automationid="splitbuttonprimary"
      >
        <i
          aria-hidden="true"
          class="ms-Icon root-105 ms-Button-icon icon-112"
          data-icon-name="ChevronDown"
        />
        <span
          aria-level="2"
          class="requirementInstructionsHeader"
          role="heading"
        >
          How to test
        </span>
      </span>
    </button>
    <div
      class="requirementInstructions collapsibleContent"
    >
      <p>
        how to test stub
      </p>
    </div>
  </div>
</DocumentFragment>
`;
